lib/schemaUtils.js

```diff
@@ -3399,17 +3401,39 @@ module.exports = {
     }, (err, res) => {
       let mismatches = [],
         mismatchObj;
+      const unmatchedSchemaVariableNames = determinedPathVariables.filter((pathVariable) => {
+        return !pathVariable._varMatched;
+      }).map((schemaPathVar) => {
+        return schemaPathVar.key;
+      });
 
       if (err) {
         return callback(err);
       }
 
       // go through required schemaPathVariables, and params that aren't found in the given transaction are errors
-      _.each(schemaPathVariables, (pathVar) => {
+      _.each(schemaPathVariables, (pathVar, index) => {
         if (!_.find(determinedPathVariables, (param) => {
           // only consider variable matching if url path variables is not allowed
           return param.key === pathVar.name && (options.allowUrlPathVarMatching || param._varMatched);
         })) {
+          let reasonCode,
+            reason,
+            key,
+            currentUnmatchedVariableInTransaction = unmatchedVariablesFromTransaction[index],
+            isInvalidValue = currentUnmatchedVariableInTransaction !== undefined;
+
+          if (unmatchedSchemaVariableNames.length > 0 && isInvalidValue) {
+            reasonCode = 'INVALID_VALUE';
```

Comment:
@erfemega In this case as we are suggesting the entire key-value pair in the suggested value, can we make the reason code as `MISSING_IN_REQUEST`? As `INVALID_VALUE` is reserved only for updating value in the path variable. (i.e. only value from key-value pair)

Reply:
Done, the suggested value, when the var name is wrong and reasonCode is INVALID_VALUE, now is only the correct variable name

Reply:
@erfemega I think you misunderstood what I was trying to say.

What I suggested here is we use reasonCode `MISSING_IN_REQUEST` and not `INVALID_VALUE` as it doesn't apply here as we suggest entire key-value pair.

Following is the response after fix. Here, as I said earlier in `transactionPath` we suggest entire variable (key-value pair) but are only suggesting name (key) of the parameter.

```
{
  "property": "PATHVARIABLE",
  "transactionJsonPath": "$.request.url.variable",
  "schemaJsonPath": "$.paths[/pets/{peterId}].get.parameters",
  "reasonCode": "INVALID_VALUE",
  "reason": "The petId path variable does not match with path variable expected (peterId) in the schema at this position",
  "suggestedFix": {
    "key": "peterId",
    "actualValue": "petId",
    "suggestedValue": "peterId"
  }
}
```

So it should be something like below if it gets you clear understanding.

```
{
  "property": "PATHVARIABLE",
  "transactionJsonPath": "$.request.url.variable",
  "schemaJsonPath": "$.paths[/pets/{peterId}].get.parameters",
  "reasonCode": "INVALID_VALUE",
  "reason": "The petId path variable does not match with path variable expected (peterId) in the schema at this position",
  "suggestedFix": {
    "key": "peterId",
    "actualValue": {
      "key": "petId",
      "value": Existing value in collection,
      "description": Existing description in collection
    },
    "suggestedValue": {
      "key": "peterId",
      "value": Existing value in collection,
      "description": Existing description in collection
    }
  }
}
```

Reply:
It's done @VShingala, now this is the mismatch when a pathVar does not match:
```
{
  property: "PATHVARIABLE",
  transactionJsonPath: "$.request.url.variable",
  schemaJsonPath: "$.paths[/pets/{peterId}].get.parameters",
  reasonCode: "MISSING_IN_REQUEST",
  reason: "The petId path variable does not match with path variable expected (peterId) in the schema at this position",
  suggestedFix: {
    key: "peterId",
    actualValue: {
      key: "petId",
      description: "(Required) The id of the pet to retrieve",
      value: "<string>",
    },
    suggestedValue: {
      key: "peterId",
      value: "consequat et",
      description: "(Required) The id of the pet to retrieve",
    },
  },
}
```